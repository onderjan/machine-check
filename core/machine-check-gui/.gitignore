# Ignore generated WASM files.
content/wasm
