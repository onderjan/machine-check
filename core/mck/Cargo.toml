[package]
name = "mck"
version = "0.1.0"
edition = "2021"
license = "MIT OR Apache-2.0"
description = "Type/operation utility crate for the formal verification tool machine-check"
rust-version = "1.75"

[dependencies]
seq-macro = "0.3.5"
