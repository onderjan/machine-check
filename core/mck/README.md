# Utility type/operation crate for machine-check

This crate is used to define concrete, abstract, etc. types and operations
for the formal verification tool [machine-check](
https://crates.io/crates/machine-check). As [machine-check](
https://crates.io/crates/machine-check) generates Rust code using these types
and operations and then builds and executes the resulting files, the name of
this crate was chosen so the generated absolute paths are as concise as possible.

# Usage and Compatibility

This crate is a utility crate for [machine-check](https://crates.io/crates/machine-check)
and should not be used on its own. No compatibility guarantees are made.

# License

This crate is licensed under Apache 2.0 License or MIT License at your discretion.
