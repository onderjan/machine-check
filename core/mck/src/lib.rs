#![doc = include_str!("../README.md")]

mod array;
mod bitvector;
mod boolean;
mod panic;
mod traits;

pub mod concr {
    pub use super::array::concr::*;
    pub use super::bitvector::concr::*;
    pub use super::boolean::concr::*;
    pub use super::panic::concr::*;
    pub use super::traits::concr::*;
}

pub mod abstr {
    pub use super::array::abstr::*;
    pub use super::bitvector::abstr::*;
    pub use super::boolean::abstr::*;
    pub use super::panic::abstr::*;
    pub use super::traits::abstr::*;
}

pub mod refin {
    pub use super::array::refin::*;
    pub use super::bitvector::refin::*;
    pub use super::boolean::refin::*;
    pub use super::panic::refin::*;
    pub use super::traits::refin::*;
}

pub mod forward {
    pub use super::traits::forward::*;
}

pub mod backward {
    pub use super::traits::backward::*;
}

pub mod misc {
    pub use super::array::light::*;
    pub use super::traits::misc::*;
}
