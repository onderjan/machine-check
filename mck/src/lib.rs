mod array;
mod bitvector;
pub mod mark;
mod traits;
mod util;

pub use array::concr::MachineArray;
pub use bitvector::concr::MachineBitvector;
pub use bitvector::mark::MarkBitvector;
pub use bitvector::three_valued::ThreeValuedBitvector;
pub use bitvector::Bitvector;

pub use traits::MachineExt;
pub use traits::MachineShift;
pub use traits::Possibility;
pub use traits::TypedCmp;
pub use traits::TypedEq;
