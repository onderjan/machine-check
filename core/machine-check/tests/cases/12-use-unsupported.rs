#[machine_check_macros::machine_description]
mod machine {
    // this is supported
    use machine_check::Bitvector;
    use machine_check::Signed as Bigga;
    use std::clone::Clone;

    // this is not
    use core::A;
    use mck;
}

fn main() {}
