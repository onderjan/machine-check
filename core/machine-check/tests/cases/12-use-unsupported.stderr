error: machine-check: Using paths not starting with 'machine_check' or 'std' not supported
 --> tests/cases/12-use-unsupported.rs:9:9
  |
9 |     use core::A;
  |         ^^^^

error: machine-check: Using paths not starting with 'machine_check' or 'std' not supported
  --> tests/cases/12-use-unsupported.rs:10:9
   |
10 |     use mck;
   |         ^^^
